src/main/java/net/neoforged/neoforge/registries/RegistryManager.java

```diff
@@ -61,6 +86,12 @@ public static void postNewRegistryEvent() {
 
         ModLoader.get().postEvent(new ModifyRegistriesEvent());
 
+        final Map<ResourceKey<Registry<?>>, Map<ResourceLocation, DataMapType<?, ?>>> dataMapTypes = new HashMap<>();
+        ModLoader.get().postEvent(new RegisterDataMapTypesEvent(dataMapTypes));
```

Comment:
Shouldn't this be fired later with the rest of the init events? The registries are still unfrozen here so I'd move this to `RegistrationEvents`.

Reply:
(un)freezing is a separate state transition. The big difference is that `RegistrationEvents` happens after configs and sided setup.